fix: [Core] update escrow function permissions

leric7 opened the pull request:
## Description

<!-- Describe the goal of this pull request. What does it change or fix? -->
Escrow functions `bulkPayout`, `storeResult`, and `complete` needs more restricted permission.
And reputation/recording oracle should not be considered super user, in general.

## Summary of changes

<!-- At a high level, what parts of the code did you change and why? -->
- Added two more modifiers. `trustedOrReputationOracle`, `trustedOrRecordingOracle`.
- Changed modifier of three functions `bulkPayout`, `storeResult`, and `complete` of escrow contract.
- Removed `recordingOracle`, and `reputationOracle` from trusted handlers list.

## How test the changes

<!-- If there are any special testing requirements, add them here -->

## Related issues
[Keywords for linking issues](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests)

<!-- Does this close any open issues? -->
Closes #558 

## Operational checklist

- [ ] All new functionality is covered by tests
- [ ] Any related documentation has been changed or added

portuu3 commented on packages/core/contracts/Escrow.sol:
```diff
@@ -47,6 +47,9 @@ contract Escrow is IEscrow, ReentrancyGuard {
     string public manifestHash;
 
+    string public intermediateResultsUrl;
+    string public intermediateResultsHash;
```
I wonder if hash of the files is actually necessary, what do you think?

leric7 replied:
Agreed. Dropped `intermediateResultsHash` param. However, want to keep both function params of `storeResults` for backwards compatibility.

portuu3 commented on packages/core/contracts/Escrow.sol:
```diff
@@ -297,20 +307,27 @@ contract Escrow is IEscrow, ReentrancyGuard {
+        _;
+    }
+
+    modifier trustedOrRecordingOracle() {
```
This modifier might not be worth since it is used only once. Check the weight of the contract, if it is lighter by removing it, we should consider to remove it

portuu3 commented on packages/examples/fortune/reputation-oracle/src/services/escrow.test.ts:
why this change?

leric7 replied:
There is a `bulkPayout` function call, and recording oracle now can't call this function.

portuu3 reviewed (changes requested):
We should update sdk and add functions if needed, like getIntermediateResultsUrl (also update docs)

leric7 commented:
@portuu3 , I resolved your comments, and added sdk functions. I don't know what you meant by updating doc, but if you were referring to updating doc for core contracts, I have another ticket for that. #560

leric7 commented:
> > @portuu3 , I resolved your comments, and added sdk functions. I don't know what you meant by updating doc, but if you were referring to updating doc for core contracts, I have another ticket for that. #560
> 
> I meant this doc https://www.notion.so/human-protocol/SDK-d312448eb4224d1886435e27897189e3

I updated.
